### DISCUSSION TITLE
Adding a quotation mark `"` in a string yields confusion

### DISCUSSION THREAD

**Discussion by StachuDotNet** [darklang/dark#4418] [Suggestions]
![dark (2)](https://user-images.githubusercontent.com/906686/186563256-bb23c7fa-ab81-4acc-a648-13a0b0165758.gif)
I expect this to result in 3 strings total being concatonated - instead, the " and ++ are simply new characters in the first string.

**Comment by StachuDotNet** [comment]
I'm not sure what's best to do here - I both want to be able to add quote marks easily, and also end the string easily.

**Reply by pbiggar** [reply]
Perhaps the right answer is to have string interpolation, and then people wouldn't do this (as much).

This is another place where it's unclear how exactly this would work, from an interaction perspective, and we would need to do that as a first step.

**Reply by StachuDotNet** [reply]
I agree.
One additional thing that would help me here would be increasing the color contrast between the text in a string and the non-string operators.
e.g. the "+" in a string vs the "++" operator.

I"m colorblind and had to really focus to figure out what was going on here - it looked as if there _were_ 3 strings being concatonated.

**Reply by pbiggar** [reply]
Great idea. Quotes in strings should also be a special color

**Reply by StachuDotNet** [reply]
Ah, that's a good idea - I was really just mentioning that the "string color" and the "operator color" could have a higher contrast.
(The 'off-white' of the string isn't far off enough from white)
